internal/bridgeapi/client.go

```diff
@@ -0,0 +1,308 @@
+/*
+Copyright 2021 Crunchy Data Solutions, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package bridgeapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/go-logr/logr"
+)
+
+var (
+	routeClusters    string = "/clusters"
+	routeDefaultRole string = "/clusters/%s/roles/default"
+)
+
+// Leave open configuration for eventual-consistency scenario
+type Client struct {
+	APITarget *url.URL
+	Log       logr.Logger
+	client    *http.Client
+}
+
+// NewClient returns a client initialized with the package logger as the
+// default logger and an uninitialized APITarget for late binding
+func NewClient() *Client {
+	client := &Client{
+		Log: pkgLog,
+	}
+	return client
+}
+
+func (c *Client) precheck() error {
+	if c.APITarget == nil {
+		return ErrorAPIUnset
+	}
+
+	// Lazy initialization
+	if c.client == nil {
+		c.client = &http.Client{}
+	}
+
+	return nil
+}
+
+// helper to set up auth with current bearer token
+func setBearer(req *http.Request) {
+	req.Header.Set("Authorization", "Bearer "+primaryLogin.activeToken)
+}
+
+func (c *Client) CreateCluster(cr CreateRequest) error {
+	if err := c.precheck(); err != nil {
+		return err
+	}
+	// TODO: Identify personal team id if not provided in request
+
+	reqPayload := new(bytes.Buffer)
+	json.NewEncoder(reqPayload).Encode(cr)
+	req, err := http.NewRequest(http.MethodPost, c.APITarget.String()+routeClusters, reqPayload)
+	if err != nil {
+		c.Log.Error(err, "during create cluster request")
+		return err
+	}
+	setBearer(req)
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		c.Log.Error(err, "during create cluster")
+		return err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusCreated:
+		return nil
+	case http.StatusBadRequest:
+		return ErrorBadRequest
+	case http.StatusConflict:
+		return ErrorConflict
+	default:
+		c.Log.Info("unrecognized return status from create call", "statusCode", resp.StatusCode)
+		return nil
+	}
+}
+
+// ClusterByName returns the cluster detail for the named cluster
+// at present, it is syntactic sugar for finding the named cluster in the
+// ListAllClusters response and retrieving its detail from the individual
+// cluster endpoint. This pivot is required as the cluster list does not
+// include the state field
+//
+// Returns a zero-value ClusterDetail and nil error when not found
+func (c *Client) ClusterByName(name string) (ClusterDetail, error) {
```

Comment:
do require a function to search cluster by id? wdyt ?

Reply:
`ClusterDetail` has the useful information contained in the cluster lists and additional information not available from the list view. I'm not sure what use case would warrant asking for a Cluster list filtered down to the ID specified when one could call `ClusterDetail` to get the relevant information for the specific cluster.

Reply:
that makes sense sounds good. thanks